docs/rules/array-bracket-spacing.md

```diff
@@ -18,26 +18,35 @@ var [x,y] = z;
 ## Rule Details
 
 This rule aims to maintain consistency around the spacing inside of array brackets, either by disallowing
-spaces inside of brackets between the brackets and other tokens or enforcing spaces. Brackets that are
-separated from the adjacent value by a new line are excepted from this rule, as this is a common pattern.
-  Object literals that are used as the first or last element in an array are also ignored.
+spaces inside of brackets between the brackets and other tokens or enforcing spaces.
 
 ## Options
 
-There are two options for this rule:
+This rule has a string option:
 
-* `"always"` enforces a space inside of array brackets
-* `"never"` enforces no space inside of array brackets (default)
+* `"never"` (default) disallows spaces inside array brackets
+* `"always"` enforces spaces inside array brackets
 
-Depending on your coding conventions, you can choose either option by specifying it in your configuration:
+This rule has built-in exceptions:
 
-```json
-"array-bracket-spacing": ["error", "always"]
-```
+* `"never"` allows newlines inside brackets of an array literal, as this is a common pattern
+* `"always"` does not require spaces in an empty array literal `[]`
+
+This rule can have an object option for exceptions to the `"never"` option:
+
+* `"singleValue": true` requires at least one space inside brackets of an array literal which has one element
+* `"objectsInArrays": true` requires at least one space between brackets of an array literal and braces of object literals which are its first or last element
```

Comment:
Good thought! In docs/rules/*.md

* 26 “brace” (sg or pl) are mostly related to blocks (including function bodies)
* 22 “curly brace” (sg or pl) are mostly related to object literals

Let’s use brace/braces consistently and add (also known as …) following the first occurrence in the docs for the 3 rules whose names include curly:

* curly
* object-curly-spacing
* template-curly-spacing

P.S. Just in the past few weeks, I added 9 occurrences of “square bracket” where “bracket” had occurred near “curly brace” so we will get rid of them!